modules/log/event_writer_console.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package log
+
+import (
+	"io"
+	"os"
+)
+
+type WriterConsoleOption struct {
+	Stderr bool
+}
+
+type eventWriterConsole struct {
+	*EventWriterBaseImpl
+}
+
+var _ EventWriter = (*eventWriterConsole)(nil)
+
+type nopCloser struct {
+	io.Writer
+}
+
+func (nopCloser) Close() error { return nil }
+
+func NewEventWriterConsole(name string, mode WriterMode) EventWriter {
+	w := &eventWriterConsole{EventWriterBaseImpl: NewEventWriterBase(name, "console", mode)}
+	opt := mode.WriterOption.(WriterConsoleOption)
+	if opt.Stderr {
+		w.OutputWriteCloser = nopCloser{os.Stderr}
+	} else {
+		w.OutputWriteCloser = nopCloser{os.Stdout}
+	}
+	return w
+}
```

Comment:
```suggestion
type nopCloser struct {}

func (nopCloser) Close() error { return nil }

func NewEventWriterConsole(name string, mode WriterMode) EventWriter {
	return &eventWriterConsole{EventWriterBaseImpl: NewEventWriterBase(name, "console", mode), OutputWriteCloser: nopCloser{}}
}
```
Should be equivalent, right?
Or do you use the Writer for anything else?

Reply:
Sorry, I don't get the point "equivalent".

`nopCloser` means it writes, but it doesn't close.

Reply:
I didn't find a `write` method, to me it looks as if its only method is `Close`.
And for that, we don't need to store the writer.
Where exactly is its `write` method?
In the `io.Writer` that doesn't seem to be used anywhere?
Or where?

Reply:
That's Golang standard.

```
type nopCloser struct {
	io.Writer
}

w :=  nopCloser{otherWriter}

// now w.Writer is otherWriter

```

Then `w.Write` equals to `w.Writer.Write`, which calls `otherWriter`